### DISCUSSION TITLE
Disable the revocation check for software certificates with JDK on the MirthAdminLauncher

### DISCUSSION THREAD

**Discussion by NoraTheExplorer** [nextgenhealthcare/connect#4679] [Q&A]
Hi,

we are using mirthconnect 3.10.X with the openJDK 13.0.1.

It works very well but on some systems the Launcher takes minutes to load the Administrator.
This is easily reproduceable on a system that has a local network connection but no access to the internet at all.

So we used wireshark to have a look into this and we've seen, that mirth is trying to connect to a OSCP server.
We assume that this is causing a delay because of timeouts, retries and so on. We actually had the exact same issues with another software.

I tried to disable the certificate check with the following options in the .vmoptions files.

So my file MirthAdminLauncher/launcher.vmoptions looks like this:
```
-Xmx512m
-Ddeployment.security.revocation.check="NO_CHECK"
-Docsp.enable=false
```
Actually I added those options to every vmoptions file in the installation folder but no changes. Yes I restarted the service, I cleared the cache but nothing seems to work.

Any recommendations?

Thanks in advance

EDIT: we just upgraded to openJDK 16.0.2 and also added the vmoptions recommended for java9 and above. I'll let you know if this is helping.

**Comment by jonbartels** [comment]
Can the `deployment.security.revocation.check` be passed as a runtime argument or must it be configured via `deployment.properties`?

A casual search turns up:
* Oracle doco https://docs.oracle.com/javase/7/docs/technotes/guides/jweb/jcp/properties.html
* OpenJDK github https://github.com/adoptium/adoptium-support/issues/129

Both suggest that that option has to be set from a specific config file, I don't see that it's tolerated as being passed as an argument. I assume I'm wrong and misreading something, what doco did you reference to choose that option?

The second link to OpenJDK, [particularly this comment ](https://github.com/adoptium/adoptium-support/issues/129#issuecomment-646481396), discusses some other options and also which ones apply to WebStart versus plain old Java. 

EDIT - Also the Launcher is closed source so I'm not sure if its using `javaws` or `java` to launch the MC client.
